frontend/catalyst/jax_tracer.py

```diff
@@ -838,15 +849,23 @@ def is_leaf(obj):
                 return_values, is_leaf=is_leaf
             )
 
-            # TODO: In order to compose transforms, we would need to recursively
-            # call apply_transform while popping the latest transform applied,
-            # until there are no more transforms to be applied.
-            # But first we should clean this up this method a bit more.
-            tapes, post_processing = apply_transform(qnode, quantum_tape, return_values_flat)
+            if isinstance(device, qml.devices.Device):
+                transform_program, _ = device.preprocess()
+            else:
+                transform_program = TransformProgram()
+
+            # We add pragma because lit test are giving qfunc directly
+            # But lit tests are not sending coverage results
+            if qnode:  # pragma: no branch
+                transform_program = qnode.transform_program + transform_program
```

Comment:
Just trying to understand how the qnode argument is used. Does the parameter really exist for the lit tests? Or would be used in regular operation as well?

(could you add the parameter to the docstring btw? :))

Reply:
There are two transform programs: one on the qnode and one on the device (preprocess).

I will add the docstrings.

Reply:
Got it 👍